tiledb/sm/query/query_macros.h

```diff
@@ -42,14 +42,16 @@ namespace tiledb::sm {
  * Returns an error status if the given Status is not Status::Ok, or
  * if the StorageManager that owns this Query has requested cancellation.
  */
```

Comment:
This comment could probably use an update. Though `process_external_cancellation` still belongs to `StorageManager`, it's now a loose claim that the SM owns the `Query` and requests the cancellation (afaict).